pkg/segment/writer/agiletree.go

```diff
@@ -41,6 +42,48 @@ const (
 	TotalMeasFns
 )
 
+var STBHolderPool [MaxConcurrentAgileTrees]*STBHolder
+
+type STBHolder struct {
+	stbPtr            *StarTreeBuilder
+	currentlyInUse    bool
+	lastUsedTimestamp time.Time
+}
+
+func GetSTB() *STBHolder {
+	atreeCounterLock.Lock()
```

Comment:
since the atree locks are just used in this file, then lets move the definition in to this file. same comment for the MaxConcurrentAgileTrees

Reply:
Addressed in the latest commit.